include/deal.II/base/template_constraints.h

```diff
@@ -599,9 +600,59 @@ template <typename T> struct EnableIfScalar<std::complex<T> >
 
 
 
+// --------------- inline functions -----------------
 
 
-// --------------- inline functions -----------------
+/**
+ * Provide an <tt>operator*</tt> that operates on mixed complex floating
+ * point types. Annoyingly, the standard library does not provide such an
+ * operator...
+ *
+ * @relates ProductType
+ */
+template <typename T, typename U>
+typename ProductType<std::complex<T>, std::complex<U> >::type
+inline
+operator*(const std::complex<T> &left, const std::complex<U> &right)
+{
+  typedef typename ProductType<std::complex<T>, std::complex<U> >::type result_type;
+  return static_cast<result_type>(left) * static_cast<result_type>(right);
+}
```

Comment:
Reason being that if the *are* defined, you will get ambiguities which operator is called (detected by the compiler or not).